doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+    {
+        InstallOptions();
+
+        /// DESIGN NOTE: --Exact from the winget install command line is implied here by virtue of the find then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up front in order to get the Package object.
+        /// DESIGN NOTE: Using PackageVersion rather than Package felt like it would be too specific here. Package and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about which version is preferred.
+        Package Package;
+        /// Optionally specifies the version from the package to install. If unspecified the version matching Package.GetLatestVersion() is used.
+        PackageVersionId PackageVersionId;
+
+        /// Specifies alternate location to install package (if supported).
+        String PreferredInstallLocation;
+        /// User or Machine. 
+        InstallScope InstallScope;
+        /// Silent, Interactive, or Default
+        InstallMode InstallMode;
+        /// Directs the logging to a log file. Callers must provide a path to a file that the Windows Package Manager package has write access to.
+        String LogOutputPath;
+        /// Continues the install even if the hash in the catalog does not match the linked installer.
+        Boolean AllowHashMismatch;
+        /// A string that will be passed to the installer. 
+        /// IMPLEMENTATION NOTE: maps to "--override" in the winget cmd line
+        String ReplacementInstallerArguments;
+
+        /// Used by a caller to correlate the install with a caller's data.
+        /// IMPLEMENTATION NOTE: This area is in flux but right now it's expected that this would allow a caller to use JSON to encode multiple arguments.
+        String AdditionalTelemetryArguments;
+        /// A string that will be passed to the source server if using a REST source
+        String AdditionalAppCatalogArguments;
+    }
+
+    runtimeclass AppInstaller
+    {
+        AppInstaller();
+
+        /// Get the available catalogs. Each source will have a separate catalog. This list does not include KnownCatalogs.
+        /// This does not open the catalog. These catalogs can be used individually or merged with GetCompositeAppCatalogAsync. 
+        /// IMPLEMENTATION NOTE: This is a list of sources returned by Windows Package Manager source list excluding the predefined sources.
+        Windows.Foundation.IAsyncOperation< Windows.Foundation.Collections.IVectorView<AppCatalog> > GetAppCatalogsAsync();
+        /// Get a built in catalog
+        [default_overload]
+        Windows.Foundation.IAsyncOperation<AppCatalog> GetAppCatalogAsync(PredefinedAppCatalog predefinedAppCatalog);
+        /// Get a catalog by a known unique identifier
+        Windows.Foundation.IAsyncOperation<AppCatalog> GetAppCatalogAsync(String catalogId);

```

Comment:
If we were doing this as a Windows API, I'd say `GetAppCatalogByIdAsync` and skip the overloading